docs/detections/rules-ui-create.asciidoc

```diff
@@ -499,7 +499,7 @@ minutes at 14:03 but it does not run until 14:04, it will run again at 14:09.
 [[rule-action-variables]]
 ==== Alert notification placeholders
 
-You can use http://mustache.github.io/[mustache syntax] to add the following placeholders to <<rule-notifications, rule action>> fields:
+You can use http://mustache.github.io/[mustache syntax] to add the following placeholders to <<rule-notifications, rule action>> fields. Any plural name (ending with the letter `s`) represents an array of values.
```

Comment:
This is not entirely correct and can be a bit fragile "rule". In general, we should be able to have any number of rule fields that end with the letter `s` which are non-arrays, and we shouldn't have artificial rules saying that a field name can't end with an `s` if it's not an array.

Specifically, at the moment we have a `{{context.rule.max_signals}}` field which doesn't comply with the above rule, because it's a single number.

I'd suggest to indicate that a field is an array per each placeholder on this page separately. For example:

```
{{context.rule.references}}: An array of string URLs referring to any information related to this rule.
```

Reply:
Also, currently we have fields that don't end with the letter `s` which are still arrays :) `{{context.rule.threat}}` is an array of objects.

Reply:
@banderror Thanks for clarifying -- I was wondering if this would hold up for all field types! 😅 I like the idea of indicating in each description whether a field is an array, as you suggest. Would you (or anyone else on the team) be able to help identify which specific fields we should update? Or is this info available somewhere else that I could reference on my own?

Reply:
@joepeeples Unfortunately we don't have any decent dev docs on the rule fields, so unless you can read TypeScript it could be difficult to "extract" this info from the codebase. If you can do it and have good IDE support for TypeScript you could try to dig into the rule schema:

https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/common/detection_engine/rule_schema/model/rule_schemas.ts

Though, let me see if I can quickly help you with this tomorrow and just post some suggestions for the placeholder descriptions.

Reply:
@banderror I can open the file in VS Code and it displays pretty well, but it might be a similar guessing game for me -- is it just any item that ends in `Array`? It'd probably be better to get the details directly from you, but there's no rush. This is more of a "nice to have" thing, especially compared to any features or enhancements for 8.6 release. Thanks!